src/DeviceListener.ts

```diff
@@ -407,6 +415,72 @@ export default class DeviceListener {
         this.displayingToastsForDeviceIds = newUnverifiedDeviceIds;
     }
 
+    /**
+     * Reports current recovery state to analytics.
+     * Checks if the session is verified and if the recovery is correctly setup (i.e all secrets known locally and in 4S).
+     * @param crypto - the crypto module
+     * @param cli - the matrix client
+     * @param secretStorageReady - if the secret storage is ready
+     * @private
+     */
+    private async reportCryptoSessionStateToAnalytics(
+        crypto: CryptoApi,
+        cli: MatrixClient,
+        secretStorageReady: boolean,
+    ): Promise<void> {
+        //
+        const crossSigningStatus = await crypto.getCrossSigningStatus();
+        // const backupInfo = await crypto.getActiveSessionBackupVersion();
+        const backupInfo = await this.getKeyBackupInfo();
+        const is4SEnabled = (await cli.secretStorage.getDefaultKeyId()) != null;
+
+        const verificationState = crossSigningStatus.publicKeysOnDevice ? "Verified" : "NotVerified";
+
+        let recoveryState: "Disabled" | "Enabled" | "Incomplete";
+        if (!is4SEnabled) {
+            recoveryState = "Disabled";
+        } else {
+            const allCrossSigningSecretsCached =
+                crossSigningStatus.privateKeysCachedLocally.masterKey &&
+                crossSigningStatus.privateKeysCachedLocally.selfSigningKey &&
+                crossSigningStatus.privateKeysCachedLocally.userSigningKey;
+            if (backupInfo != null) {
+                // there is a backup check that all secrets are stored in 4s and known locally
+                // if they are not, the backup is incomplete
+                const backupPrivateKeyIsInCache = (await crypto.getSessionBackupPrivateKey()) != null;
+                if (secretStorageReady && allCrossSigningSecretsCached && backupPrivateKeyIsInCache) {
+                    recoveryState = "Enabled";
+                } else {
+                    recoveryState = "Incomplete";
+                }
+            } else {
+                // no backup just consider cross-signing secrets
+                if (allCrossSigningSecretsCached && crossSigningStatus.privateKeysInSecretStorage) {
```

Comment:
could we make this more consistent with the check when there is a backup?

```suggestion
                if (secretStorageReady && allCrossSigningSecretsCached) {
```

(I think the two are equivalent?)

Reply:
Yes there is no backup so `secretStorageReady` checks for cross signing secrets to be in 4S.
Updated and added a new test